feat: add concurrency to workflow

virtualroot opened the pull request:
Allow only one workflow at that time.

If a second workflow is triggered, it will be put in pending state, and only start once the first workflow finish.
If the workflow is triggered a third time while the first one is running, the second workflow is canceled, and the third takes over the pending state.

Fix #163.

virtualroot committed:
feat: add concurrency to workflow

Allow only one workflow at that time.

If a second workflow is triggered, it will be put in pending state, and
only start once the first workflow finish.
If the workflow is triggered a third time while the first one is
running, the second workflow is canceled, and the third takes over the
pending state.

Fix #163.

Signed-off-by: Alejandro Lazaro <alazaro@werpo.com.ar>

cam72cam commented on .github/workflows/bump-and-publish-versions-prod.yml:
```diff
@@ -5,6 +5,9 @@ on:
   workflow_dispatch:
 
+concurrency:
+  group: ${{ github.head_ref }}
```
Bump versions can put new commits on main, which subsequent invocations would run under. Do we want this to use a static string instead?

virtualroot replied:
@cam72cam 

I chose the dynamic value since I noticed the `workflow_dispatch`.
If there are no plans to do runs from branches other than the main, I can hardcode.

cam72cam replied:
I just took a closer look athttps://docs.github.com/en/actions/using-jobs/using-concurrency#example-using-a-fallback-value.

We may want to run it on branches as validation at some point, but for now a global lock should suffice.  We should probably add a brief comment about this conversation in the file.

virtualroot replied:
I changed the value to `main`, added a comment in the commit message, and linked to this discussion.

virtualroot committed:
fix: use global lock

We may want to run it on branches as validation at some point, but for now a global lock should suffice.

Signed-off-by: Alejandro Lazaro <alazaro@werpo.com.ar>

virtualroot commented:
@cube2222 #163 was opened by you. Please have a look and merge if it's okay.
